src/main/java/io/citrine/jcc/search/core/query/AbstractFieldQuery.java

```diff
@@ -0,0 +1,198 @@
+package io.citrine.jcc.search.core.query;
+
+import io.citrine.jcc.search.analysis.query.Analysis;
+import io.citrine.jcc.util.ListUtil;
+import io.citrine.jcc.util.MapUtil;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Base class for all queries against simple fields.
+ *
+ * @author Kyle Michel
+ */
+public abstract class AbstractFieldQuery implements HasLogic, HasWeight, HasSimple {
+
+    @Override
+    public AbstractFieldQuery setLogic(final Logic logic) {
+        this.logic = logic;
+        return this;
+    }
+
+    @Override
+    public Logic getLogic() {
+        return this.logic;
+    }
+
+    @Override
+    public AbstractFieldQuery setWeight(final Double weight) {
+        this.weight = weight;
+        return this;
+    }
+
+    @Override
+    public Double getWeight() {
+        return this.weight;
+    }
+
+    @Override
+    public AbstractFieldQuery setSimple(final String simple) {
+        this.simple = simple;
+        return this;
+    }
+
+    @Override
+    public String getSimple() {
+        return this.simple;
+    }
+
+    @Override
+    public AbstractFieldQuery setSimpleWeight(final Map<String, Double> simpleWeight) {
+        this.simpleWeight = simpleWeight;
+        return this;
+    }
+
+    @Override
+    public AbstractFieldQuery addSimpleWeight(final Map<String, Double> simpleWeight) {
+        this.simpleWeight = MapUtil.add(simpleWeight, this.simpleWeight);
+        return this;
+    }
+
+    @Override
+    public AbstractFieldQuery addSimpleWeight(final String field, final Double weight) {
+        this.simpleWeight = MapUtil.add(field, weight, this.simpleWeight);
+        return this;
+    }
+
+    @Override
+    public Map<String, Double> getSimpleWeight() {
```

Comment:
Why do you need to override a getter?

Reply:
good catch, removing

Reply:
Oh nevermind. It is because the method is defined in an interface so it does not have an implementation yet.